nifi-commons/nifi-socket-utils/src/main/java/org/apache/nifi/io/socket/SSLContextFactory.java

```diff
@@ -49,7 +55,7 @@
     private final KeyManager[] keyManagers;
     private final TrustManager[] trustManagers;
 
-    public SSLContextFactory(final NiFiProperties properties) throws NoSuchAlgorithmException, CertificateException, FileNotFoundException, IOException, KeyStoreException, UnrecoverableKeyException {
+    public SSLContextFactory(final NiFiProperties properties) throws NoSuchAlgorithmException, CertificateException, FileNotFoundException, IOException, KeyStoreException, UnrecoverableKeyException, InvalidAlgorithmParameterException {
```

Comment:
Is there an implementation of this constructor / static factory method in the application which can create an SSLContext that does not require the application to have HTTPS enabled? For example, an instance of NiFi running in HTTP mode which needs to use `InvokeHTTP` to communicate with an HTTPS remote endpoint?